### PR TITLE
Drupal 10 support

### PR DESCRIPTION
**Pull request by chesn0k**

### CODE REVIEW

**Comment by chesn0k** [comment]
@Niklan, let's discuss the PHPCS and PHPStan configuration for this module

**Comment by Niklan** [comment]
First, dropping Drupal 9- in minor release is extreme even for an alpha module. I propose to create a 3.x branch for these changes. It will be easy to migrate for 2.x version. Also, we still need to keep supporting core ^9.5 for smoother upgrade for projects. We can drop 9.5 in minor release when it wont be supported.

For PHPCS we better to use https://github.com/Chi-teck/drupal-coder-extension/tree/2.x

For PHPStan, L5 should be a minimum requirement.

**Comment by Niklan** [comment]
Also, we need to create tests for this module. I can do that.

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "license": "GPL-2.0+",
     "homepage": "https://www.drupal.org/project/cloner",
     "require": {
-        "drupal/core": "^8.8 || ^9"
+        "php": ">=8.1",
+        "drupal/core": "^10"
```
Comment (Niklan) [inline composer.json]: Should be `^9.5 || ^10`

File: cloner.info.yml [hunk 2]
```diff
@@ -2,5 +2,4 @@ name: Cloner
 type: module
 description: The entity clone plugin system.
 package: Custom
-core_version_requirement: ^8.8 || ^9
-php: 7.3
+core_version_requirement: ^10
```
Comment (Niklan) [inline cloner.info.yml]: Should be ^9.5 || ^10

File: modules/cloner_examples/cloner_examples.info.yml [hunk 3]
```diff
@@ -2,6 +2,6 @@ name: Cloner Examples
 type: module
 description: The entity clone plugin examples.
 package: Development
-core_version_requirement: ^8 || ^9
+core_version_requirement: ^10
```
Comment (Niklan) [inline modules/cloner_examples/cloner_examples.info.yml]: Should be ^9.5 || ^10

File: cloner.info.yml [hunk 4]
```diff
@@ -2,5 +2,4 @@ name: Cloner
 type: module
 description: The entity clone plugin system.
 package: Custom
-core_version_requirement: ^8.8 || ^9
-php: 7.3
```
Comment (Niklan) [inline cloner.info.yml:6]: Should be `8.1`

**Pull request merged**